python/ray/tune/execution/trial_runner.py

```diff
@@ -977,7 +977,7 @@ def _on_training_result(self, trial, result):
     def _post_process_on_training_saving_result(self, trial):
         # `self._queued_trial_decisions` now contains a final decision
         # based on all results
-        if trial not in self._cached_trial_decisions:
+        if trial.trial_id not in self._cached_trial_decisions:
             final_decision = self._queued_trial_decisions.pop(trial.trial_id, None)
             if final_decision:
                 self._execute_action(trial, final_decision)
```

Comment:
I actually think we can get rid if line 980 altogether (previously it was always True)
```suggestion
        final_decision = self._queued_trial_decisions.pop(trial.trial_id, None)
        if final_decision:
            self._execute_action(trial, final_decision)
```